Sources/StreamChat/WebSocketClient/EventMiddlewares/EventMiddleware_Tests.swift

```diff
@@ -2,16 +2,17 @@
 // Copyright © 2021 Stream.io Inc. All rights reserved.
 //
 
+import CoreData
 @testable import StreamChat
 import XCTest
 
 class EventMiddleware_Tests: XCTestCase {
     /// A test middleware that can be initiated with a closure/
     struct ClosureBasedMiddleware: EventMiddleware {
-        let closure: (_ event: Event, _ completion: @escaping (Event?) -> Void) -> Void
+        let closure: (_ event: Event, _ session: DatabaseSession, _ completion: @escaping (Event?) -> Void) -> Void
         
-        func handle(event: Event, completion: @escaping (Event?) -> Void) {
-            closure(event, completion)
+        func handle(event: Event, session: DatabaseSession, completion: @escaping (Event?) -> Void) {
+            closure(event, session, completion)
```

Comment:
Maybe we can use `EventMiddlewareMock` here and get rid of `ClosureBasedMiddleware`?